wcfsetup/install/files/lib/system/file/upload/UploadHandler.class.php

```diff
@@ -0,0 +1,366 @@
+<?php
+namespace wcf\system\file\upload;
+use wcf\system\exception\ImplementationException;
+use wcf\system\SingletonFactory;
+use wcf\system\WCF;
+
+/**
+ * Handles uploads for files.
+ *
+ * @author	Joshua Ruesweg
+ * @copyright	2001-2018 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\System\File\Upload
+ * @since       5.2
+ */
+class UploadHandler extends SingletonFactory {
+	/**
+	 * Session variable name for the file storage.
+	 * @var string
+	 */
+	const UPLOAD_HANDLER_SESSION_VAR = 'file_upload_handler_storage';
+	
+	/**
+	 * Contains the registered upload fields. 
+	 * 
+	 * @var UploadField[]
+	 */
+	protected $fields = [];
+	
+	/**
+	 * Registers a UploadField.
+	 * 
+	 * @param       UploadField     $field
+	 */
+	public function registerUploadField(UploadField $field) {
+		if (isset($this->fields[$field->getFieldId()])) {
+			throw new \InvalidArgumentException('UploadField with the id "'. $field->getFieldId() .'" is already registered.');
```

Comment:
Document that an exception can be thrown and explain briefly why/when it can be thrown. (Check all methods)